File: daily/Arrays/633.Sum-of-square-numbers.py
import math


class Solution:
    def judgeSquareSum(self, c: int) -> bool:
        left = 0
        right = int(math.ceil(math.sqrt(c)))
        while left <= right:
            s = left * left + right * right
            if s == c:
                return True
            if s < c:
                left += 1
            else:
                right -= 1
        return False

File: daily/Arrays/633.Sum-of-square-numbers.go
package daily

import "math"

func judgeSquareSum(c int) bool {
	left := 0
	right := int(math.Ceil(math.Sqrt(float64(c))))

	for left <= right {
		result := left*left + right*right
		if result == c {
			return true
		} else if result < c {
			left++
		} else {
			right--
		}
	}
	return false
}
